R/model.R

```diff
@@ -506,6 +506,114 @@ compile_method <- function(quiet = TRUE,
+#' @aliases syntax_check
+#' @family CmdStanModel methods
+#'
+#' @description The `$syntax_check()` method of a [`CmdStanModel`] object
+#'   checks the Stan syntax of the model and returns `TRUE` if the compilation
+#'   succeeds. In case of a syntax error, an error is thrown.
+#'   
+#' @section Usage:
+#'   ```
+#'   $syntax_check(
+#'     quiet = TRUE,
+#'     include_paths = NULL,
+#'     stanc_options = list()
+#'   )
+#'   ```
+#'
+#' @section Arguments:
+#'   * `quiet`: (logical) Should the compile command be printed.
+#'   * `include_paths`: (character vector) Paths to directories where Stan
+#'   should look for files specified in `#include` directives in the Stan
+#'   program.
+#'   * `stanc_options`: (list) Any Stan-to-C++ transpiler options to be used
+#'   when compiling the model. See the **Examples** section below as well as the
+#'   `stanc` chapter of the CmdStan Guide for more details on available options:
+#'   https://mc-stan.org/docs/cmdstan-guide/stanc.html.
+#'
+#' @section Value: The `$syntax_check()` method returns `TRUE` if the model
+#'   is valid.
+#'
+#' @template seealso-docs
+#'
+#' @examples
+#' \dontrun{
+#' file <- file.path(cmdstan_path(), "examples/bernoulli/bernoulli.stan")
+#'
+#' mod <- cmdstan_model(file, compile = FALSE)
+#' mod$syntax_check()
+#' }
+#'
+NULL
+
+syntax_check_method <- function(quiet = TRUE,
+                           include_paths = NULL,
+                           stanc_options = list()) {
+  if (length(stanc_options) == 0 && !is.null(private$precompile_stanc_options_)) {
+    stanc_options <- private$precompile_stanc_options_
+  }
+  if (is.null(include_paths) && !is.null(private$precompile_include_paths_)) {
+    include_paths <- private$precompile_include_paths_
+  }
+  
+  model_name <- sub(" ", "_", paste0(strip_ext(basename(self$stan_file())), "_model"))
+
+  message("Running syntax check of the Stan program...")
+
+  temp_hpp_file <- tempfile(pattern = "model-", fileext = ".hpp")
+  stanc_options[["o"]] <- temp_hpp_file
+
+  stancflags_val <- ""
+  if (!is.null(include_paths)) {
+    checkmate::assert_directory_exists(include_paths, access = "r")
+    include_paths <- absolute_path(include_paths)
+    include_paths <- paste0(include_paths, collapse = ",")
+    if (cmdstan_version() >= "2.24") {
+      include_paths_flag <- " --include-paths="
+    } else {
+      include_paths_flag <- " --include_paths="
+    }
+    stancflags_val <- paste0(stancflags_val, include_paths_flag, include_paths, " ")
+  }
+
+  if (is.null(stanc_options[["name"]])) {
+    stanc_options[["name"]] <- model_name
+  }
+
+  stanc_built_options = c()
+  for (i in seq_len(length(stanc_options))) {
+    option_name <- names(stanc_options)[i]
+    if (isTRUE(as.logical(stanc_options[[i]]))) {
+      stanc_built_options <- c(stanc_built_options, paste0("--", option_name))
+    } else {
+      stanc_built_options <- c(stanc_built_options, paste0("--", option_name, "=", stanc_options[[i]]))
+    }
+  }
```

Comment:
If possible can you add a test just making sure that this works or at least doesn't error?

Reply:
stanc_options are used always for at least the name, so this works and is already being tested. Will add a test where we specify an additional flag.